build/v1/types.go

```diff
@@ -86,6 +86,16 @@ type CommonSpec struct {
 	// are ignored.
 	// +optional
 	NodeSelector OptionalNodeSelector `json:"nodeSelector" protobuf:"bytes,9,name=nodeSelector"`
+
+	// mountTrustedCA bind mounts the cluster's trusted certificate authorities, as defined in
+	// the cluster's proxy configuration, into the build. This lets processes within a build trust
+	// components signed by custom PKI certificate authorities, such as private artifact
```

Comment:
it's not appropriate to indicate that a proxy CA is used to validate private artifact repositories.  that is more of a bug than a feature.

Reply:
To piggy back on Ben's comment - unfortunately we explicitly tell customers today to use the Proxy "trusted CA" as the trusted CA for the whole cluster [1]. Connecting to a private Artifactory or Nexus repository with a corporate cert is perhaps the most common reason why a customer would use this feature.

https://docs.openshift.com/container-platform/4.6/networking/configuring-a-custom-pki.html

Reply:
> unfortunately we explicitly tell customers today to use the Proxy "trusted CA" as the trusted CA for the whole cluster 

it's not unfortunate, it was a deliberate choice that we built a mechanism to distribute the CAs generically, not specifically for proxy.

The mistake was that you have to tie the trusted CA bundle configmap to the proxy config (via a ref in the proxyconfig to the ca bundle CM) to make that happen, which makes it seem like the CAs are explicitly tied to proxy.

we said at the time we did this that if we got pushback/uses cases from customers to more finely subdivide their CA bundles, we would add additional mechanisms/configs to support that subdivision.  So far i haven't heard of any.